lib/src/mistralai_client_dart_base.dart

```diff
@@ -19,8 +24,38 @@ class MistralAIClient {
   final int maxRetries;
   final http.Client _client;
 
+  Future<String> _getRequest<T>(String endpoint) async {
+    final headers = <String, String>{
+      'Accept': 'application/json',
+      'Content-Type': 'application/json',
+      'Authorization': 'Bearer $apiKey',
+    };
+
+    try {
+      final response = await _client
+          .get(Uri.parse('$baseUrl$endpoint'), headers: headers)
+          .timeout(timeout);
+
+      if (response.statusCode != 200) {
+        throw Exception('Get request failed: ${response.statusCode}');
+      }
+
+      return response.body;
+    } catch (e) {
+      throw Exception('Get request failed: $e');
+    }
+  }
+
+  /// Returns a list of the available models [ModelsList] 
+  /// from the mistral AI API.
+  ///
+  /// It uses [list models endpoint](https://api.mistral.ai/v1/models) from the mistral AI API.
   Future<ModelsList> listModels() async {
```

Comment:
It looks a bit weird that we return the list as a single `ModelsList` 🤔 Can't we change it to `List<Model>`, which is more idiomatic for Dart?

Reply:
Okay, that makes sense 🤔 So maybe to highlight that we are actually returning the exact copy of the API response, maybe let's rename the `ModelsList` to something that will indicate it more clearly? Something like `ListModelsResult` or `ListModelsResponse`? 🤔

Reply:
We discussed using something like `ListModelResponse` but the `Response` word can be misleading because it is just a body from http.Response object. But `ListModelsResult` is ok for me.

Reply:
We agreed on `ListModelsResult` on a call 😉